format_spec.md

```diff
@@ -0,0 +1,399 @@
+
+```jsonc
+"successors": [
+  {
+    "id": "first-successor"
+  },
+  {
+    "id": "second-successor",
+    "src": "https://example.com/second-successor.modip.json"
+    // This successor is hosted externally, and not on the same host as the parent project
+    // See the `src` field in dependencies for more info.
+  }
+]
+```
+
+In the example above, the project `first-successor` was released first. After the release of `first-successor`, the project `second-successor` was released.
+
+The successor and predecessor projects should, ideally, be the same project, just under a different ID. For example, the project may have been forked and an updated version was created by a different person.
+
+---
+
+## Version Information
+
+Projects can contain a `versions` array, which lists Version Objects. These objects are stored in chronological order from most recent to least recent.
+
+## Version Objects
+
+Version Objects contain fields relating to a certain version of a Project. Below are all the fields that a Version Object may contain.
+
+### `id`
+
+The ID of this version. There is no required format for how to format a version ID, other than that they CANNOT include spaces.
+
+---
+
+### `name`
+
+The human-readable name of this version. It is RECOMMENDED to follow Semantic Versioning, but it is not required. Spaces MAY be used in this name.
+
+---
+
+### `semver` (optional)
+
+An optional Semantic Versioning compatible version number. It is HIGHLY RECOMMENDED that this field only be present when a version's name conforms to SemVer. This field CAN be present when the version's name does not conform to SemVer, however it's recommended to only be present if the name conforms.
+
+Launchers can use this field to compare different versions. However, because many Minecraft mods do not conform to SemVer, it's recommended for launchers to compare using other fields, such as `releaseDate`.
+
+---
+
+### `releaseDate`
+
+The release date of this version. It MUST be stored as an ISO-8601 conforming String. This MUST include UTC time at the end. A valid example is `2020-01-01T12:00:00Z`. This example date is the 1st of January, 2020 at 12:00:00 UTC. Spaces and time zone offsets (e.g. `+01:00`) CANNOT be used. The `Z` at the end of the string MUST be included and it MUST be capitalized. The `T` separating the date and time MUST be included and it MUST be capitalized. Other values, such as `2020-W32` are NOT allowed. The ONLY allowed format is the one demonstrated in the example.
+
+If the time of a release is unknown, it SHOULD default to `12:00:00Z`. If the date is completely unknown or hosts do not wish to provide inaccurate data, the value `unknown` CAN be passed.
+
+---
+
+### `changelog`
+
+The changelog of this version. The changelog MAY contain HTML. It follows the same rules as `description` for what is allowed.
+
+---
+
+### `allowed` (optional)
+
+Whether or not this version is allowed. **MUST be a condition.**
```

Comment:
What does this mean?

Reply:
It must be a condition value, meaning that it cannot be a raw boolean.
Valid Example:
```json
"allowed": {
  "environment": {
    "client": true,
    "server": false
  }
}
```
Invalid Example:
```json
"allowed": false
```
I should clarify it.

Reply:
Yes but what does the entire field mean?

Reply:
It indicates if a version is “allowed” to be installed, based on the condition. A launcher should obey this by either refusing to install the mod, or requesting confirmation from the user. The main purpose of the field is for client-side mods that shouldn’t be on the server.